centreon/src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php

```diff
@@ -241,7 +241,7 @@ public function canRefreshToken(): bool
      * @throws ConfigurationException
      * @throws SSOAuthenticationException
      */
-    public function authenticateOrFail(?string $authorizationCode, string $clientIp): void
+    public function authenticateOrFail(?string $authorizationCode, string $clientIp): string
```

Comment:
The name of the method does not correspond to its main intention.

Reply:
This whole code is not well designed, the wording didn't seemed important at this point.
**How would you named it ?**